CONTRIBUTING.md

```diff
@@ -0,0 +1,66 @@
+# Contributing
+
+Implementing and maintaining a protocol stack is a lot of work, therefore any
+help is appreciated, from creating issues, to contributing documentation, fixing
+issues and adding new features.
+
+Please follow the best-practice contribution guidelines as mentioned below when
+submitting any changes.
+
+### Contributions License Agreement
+
+Before contributing, please read the [Contributions License Agreement](./CLA.md).
+Every contributor must sign this agreement before any code is merged into the
+repository.
+
+### Git
+
+Please, do your changes with the develop branch in your forked repo
+and send us your pull request from your own forked repo feature-branches
+or fix-branches to the develop branch of our repo!
+
+In the case that Commit rights were granted to Contributors the following basic ground-rules should be applied:
+
+* No --force pushes or modifying the Git history in any way.
+* Non-master branches ought to be used for ongoing work.
+* External API changes and significant modifications ought to be subject to an internal pull-request to solicit feedback from other contributors.
```

Comment:
Maybe this should go in his own section since it is not directly related to git, maybe called "Guidelines for contributions" or something like this.

This section will talk about different changes / fixes and when asking for a group approval from the other contributors.

This section can also contain the "No external libraries" guideline.
The reasons for this are: more portable, less maintenance, smaller compiled code size.

Reply:
kk